cmd/commands/git-source.go

```diff
@@ -189,53 +199,67 @@ func NewGitSourceCreateCommand() *cobra.Command {
 				RuntimeName:         args[0],
 				CreateDemoResources: false,
+				Include:             include,
+				Exclude:             exclude,
+				Flow:                flow,
 			})
 		},
 	}
 
 	cmd.Flags().BoolVar(&createRepo, "create-repo", false, "If true, will create the specified git-source repo in case it doesn't already exist")
+	cmd.Flags().StringVar(&include, "include", "", "include")
+	cmd.Flags().StringVar(&exclude, "exclude", "", "exclude")
 
 	insCloneOpts = apu.AddCloneFlags(cmd, &apu.CloneFlagsOptions{CloneForWrite: true})
 	gsCloneOpts = apu.AddCloneFlags(cmd, &apu.CloneFlagsOptions{
 		Prefix:   "git-src",
 		Optional: true,
 	})
 
+	flow = store.Get().GsCreateFlow
+
 	return cmd
 }
 
 func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error {
-	// upsert git-source repo
-
-	gsRepo, gsFs, err := opts.GsCloneOpts.GetRepo(ctx)
+	version, err := getRuntimeVersion(ctx, opts.RuntimeName)
 	if err != nil {
-		return fmt.Errorf("failed to clone git-source repo: %w", err)
+		return err
 	}
 
-	if opts.CreateDemoResources {
-		if err := createDemoResources(ctx, opts, gsRepo, gsFs); err != nil {
-			return fmt.Errorf("failed to create git-source demo resources: %w", err)
-		}
+	if opts.Flow == store.Get().InstallationFlow {
+		return legacyGitSourceCreate(ctx, opts)
+	}
+
+	if version.LessThan(versionOfGitSourceByAppProxyRefactor) {
+		log.G(ctx).Infof("runtime \"%s\" is using a depracated git-source api. Versions %s and up use the app-proxy for this command", opts.RuntimeName, minAddClusterSupportedVersion)
+		return legacyGitSourceCreate(ctx, opts)
 	} else {
-		if err := createPlaceholderIfNeeded(ctx, opts, gsRepo, gsFs); err != nil {
-			return fmt.Errorf("failed to create a git-source placeholder: %w", err)
+		appProxy, err := cfConfig.NewClient().AppProxy(ctx, opts.RuntimeName, store.Get().InsecureIngressHost)
+		if err != nil {
+			return err
 		}
-	}
 
-	appDef := &runtime.AppDef{
-		Name: opts.GsName,
-		Type: application.AppTypeDirectory,
-		URL:  opts.GsCloneOpts.Repo,
-	}
+		appSpecifier := opts.GsCloneOpts.Repo
+		isInternal := util.StringIndexOf(store.Get().CFInternalGitSources, opts.GsName) > -1
+
+		err = appProxy.AppProxyGitSources().Create(ctx, &appProxyModel.CreateGitSourceInput{
+			AppName:       opts.GsName,
+			AppSpecifier:  appSpecifier,
+			DestServer:    store.Get().InCluster,
+			DestNamespace: opts.RuntimeName,
+			IsInternal:    &isInternal,
+			Include:       &opts.Include,
+			Exclude:       &opts.Exclude,
+		})
 
-	appDef.IsInternal = util.StringIndexOf(store.Get().CFInternalGitSources, appDef.Name) > -1
+		if err != nil {
+			return fmt.Errorf("failed to create git-source: %w", err)
+		}
 
-	if err := appDef.CreateApp(ctx, nil, opts.InsCloneOpts, opts.RuntimeName, store.Get().CFGitSourceType, opts.Include, ""); err != nil {
-		return fmt.Errorf("failed to create git-source application. Err: %w", err)
+		log.G(ctx).Infof("Successfully created git-source: \"%s\"", opts.GsName)
```

Comment:
there's an issue here.
gs creation can fail in app-proxy, but still print out this success message.
it's because current app-proxy just logs the error, doesn't return it.
we need to talk about how to address that.

Reply:
wouldn't a failure in app-proxy cause the graphql request to fail?

Reply:
I noticed cases where it doesn't, for example if there's a problem with the `.exec` (e.g it can't find the ap bin), it prints the error in app-proxy, but doesn't reflect that in the cli